[minizip-ng] add new port

prateek9623 opened the pull request:
**Describe the pull request**

- What does your PR fix? Fixes #12917

- Which triplets are supported/not supported? Have you updated the CI baseline?
Except for UWP, all officials should be supported.
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
yes

prateek9623 committed:
[minizip-ng] add new port

autoantwort commented:
Run `./vcpkg format-manifest ports/minizip-ng/vcpkg.json` to make the pipeline happy :)

prateek9623 commented:
> ./vcpkg format-manifest ports/minizip-ng/vcpkg.jso

Thanks

autoantwort commented:
I think you don't support uwp, so add `"supports": "!uwp"` to your vcpkg.json file and run the format thing again :)

autoantwort commented:
Now a final `./vcpkg x-add-version --overwrite-version minizip-ng` and the PR should be ready :)

autoantwort commented:
I have looked at the linked Issue and found https://github.com/oswjk/vcpkg-minizip-fork, is there a reason why you have not reused these files?

prateek9623 commented:
> I have looked at the linked Issue and found https://github.com/oswjk/vcpkg-minizip-fork, is there a reason why you have not reused these files?

those files are pretty old, we don't require those patches now.

autoantwort commented:
Ok, I asked because https://github.com/oswjk/vcpkg-minizip-fork/blob/master/CONTROL supported a lot more features than this PR

prateek9623 commented:
> Ok, I asked because https://github.com/oswjk/vcpkg-minizip-fork/blob/master/CONTROL supported a lot more features than this PR

Ya, I know, but I don't that should we support it or not. I have linked this PR to the minizip-ng repo owner https://github.com/zlib-ng/minizip-ng/issues/540

nmoinvaz commented on ports/minizip-ng/vcpkg.json:
```diff
@@ -0,0 +1,17 @@
+    "liblzma",
+    {
+      "name": "openssl",
+      "platform": "!windows"
```
OpenSSL is not needed on Apple as it can use CommonCrypto.

nmoinvaz commented on ports/minizip-ng/vcpkg.json:
```diff
@@ -0,0 +1,17 @@
+  "version-string": "3.0.1",
+  "description": "minizip-ng is a zip manipulation library written in C that is supported on Windows, macOS, and Linux.",
+  "homepage": "https://github.com/zlib-ng/minizip-ng",
+  "supports": "!uwp",
```
Does it not work on UWP? I have not tested it personally.

prateek9623 replied:
ya CI was failing on it

autoantwort replied:
No, the error was:
```
Microsoft (R) Build Engine version 16.8.2+25e4d540b for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 4/5/2021 3:39:57 PM.
     1>Project "D:\buildtrees\minizip-ng\x64-uwp-dbg\install.vcxproj" on node 1 (default targets).
     1>Project "D:\buildtrees\minizip-ng\x64-uwp-dbg\install.vcxproj" (1) is building "D:\buildtrees\minizip-ng\x64-uwp-dbg\ZERO_CHECK.vcxproj" (2) on node 1 (default targets).
     2>PrepareForBuild:
         Creating directory "x64\Debug\ZERO_CHECK\".
         Creating directory "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\".
         Creating directory "Generated Files\".
       InitializeBuildStatus:
         Creating "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
       CustomBuild:
         Checking Build System
       FinalizeBuildStatus:
         Deleting file "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild".
         Touching "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate".
     2>Done Building Project "D:\buildtrees\minizip-ng\x64-uwp-dbg\ZERO_CHECK.vcxproj" (default targets).
     1>Project "D:\buildtrees\minizip-ng\x64-uwp-dbg\install.vcxproj" (1) is building "D:\buildtrees\minizip-ng\x64-uwp-dbg\ALL_BUILD.vcxproj" (3) on node 2 (default targets).
     3>Project "D:\buildtrees\minizip-ng\x64-uwp-dbg\ALL_BUILD.vcxproj" (3) is building "D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj" (4) on node 2 (default targets).
     4>PrepareForBuild:
         Creating directory "minizip-ng.dir\Debug\".
         Creating directory "D:\buildtrees\minizip-ng\x64-uwp-dbg\Debug\".
         Creating directory "minizip-ng.dir\Debug\minizip-ng.tlog\".
       GetInstalledSDKLocations:
         Searching for SDKs targeting "UAP, 10.0.19041.0".
         Searching for SDKs targeting "Windows, 8.1".
       ResolveSDKReferences:
           Reading SDK manifest file "C:\Program Files (x86)\Microsoft SDKs\Windows Kits\10\ExtensionSDKs\Microsoft.VCLibs\14.0\SDKManifest.xml".
           Targeted configuration and architecture "Debug|x64"
           Could not find "FrameworkIdentity" attribute "FrameworkIdentity-Debug-x64" in the SDK manifest. 
           Found "FrameworkIdentity" attribute "Name = Microsoft.VCLibs.140.00.Debug, MinVersion = 14.0.27810.0, Publisher = 'CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US'" in the SDK manifest.
           Found "APPX" location attribute "AppX-Debug-x86=.\AppX\Debug\x86\Microsoft.VCLibs.x86.Debug.14.00.appx" in the SDK manifest.
           Found "APPX" location attribute "AppX-Debug-x64=.\AppX\Debug\x64\Microsoft.VCLibs.x64.Debug.14.00.appx" in the SDK manifest.
           Found "APPX" location attribute "AppX-Debug-ARM=.\AppX\Debug\ARM\Microsoft.VCLibs.ARM.Debug.14.00.appx" in the SDK manifest.
           Found "APPX" location attribute "AppX-Debug-ARM64=.\AppX\Debug\ARM64\Microsoft.VCLibs.ARM64.Debug.14.00.appx" in the SDK manifest.
       ExpandSDKReferences:
         Enumerating SDK Reference "Microsoft.VCLibs, Version=14.0" from "C:\Program Files (x86)\Microsoft SDKs\Windows Kits\10\ExtensionSDKs\Microsoft.VCLibs\14.0\".
       InitializeBuildStatus:
         Creating "minizip-ng.dir\Debug\minizip-ng.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
       CustomBuild:
         Building Custom Rule D:/buildtrees/minizip-ng/src/3.0.1-4535e59d51.clean/CMakeLists.txt
       ClCompile:
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\bin\HostX64\x64\CL.exe /c /I"D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean" /I"D:\installed\x64-uwp\include" /I"D:\buildtrees\minizip-ng\x64-uwp-dbg\\" /I"Generated Files\\" /I"minizip-ng.dir\Debug\\" /Zi /nologo /W3 /WX- /diagnostics:column /sdl /MP /Od /Ob0 /Oi /Oy- /D WIN32 /D _WINDOWS /D _DEBUG /D HAVE_STDINT_H /D HAVE_INTTYPES_H /D NO_FSEEKO /D HAVE_ZLIB /D HAVE_BZIP2 /D HAVE_LZMA /D LZMA_API_STATIC /D HAVE_ZSTD /D _CRT_SECURE_NO_DEPRECATE /D MZ_WINRT_API /D MZ_ZIP_SIGNING /D HAVE_PKCRYPT /D HAVE_WZAES /D "CMAKE_INTDIR=\"Debug\"" /D MZ_EXPORTS /D _WINDLL /D _UNICODE /D UNICODE /D WINAPI_FAMILY=WINAPI_FAMILY_APP /D __WRL_NO_DEFAULT_LIB__ /Gm- /RTC1 /MDd /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"minizip-ng.dir\Debug\\" /Fd"minizip-ng.dir\Debug\vc142.pdb" /Gd /TC /FU"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\lib\x86\store\references\platform.winmd" /analyze- /errorReport:queue  /utf-8 "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_os.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm_buf.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm_mem.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm_split.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_zip.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_zip_rw.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm_zlib.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm_bzip.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm_lzma.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm_zstd.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_os_win32.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm_os_win32.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm_pkcrypt.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_strm_wzaes.c" "D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_compat.c"
         mz_crypt.c
         mz_os.c
         mz_strm.c
         mz_strm_buf.c
         mz_strm_mem.c
         mz_strm_split.c
         mz_zip.c
         mz_zip_rw.c
         mz_strm_zlib.c
         mz_strm_bzip.c
         mz_strm_lzma.c
         mz_strm_zstd.c
         mz_os_win32.c
         mz_strm_os_win32.c
         mz_crypt_win32.c
         mz_strm_pkcrypt.c
         mz_strm_wzaes.c
         mz_compat.c
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(25,33): warning C4013: 'CryptAcquireContext' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(27,32): warning C4013: 'CryptGenRandom' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(50,25): warning C4013: 'CryptDestroyHash' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(80,33): warning C4013: 'CryptCreateHash' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(96,27): warning C4013: 'CryptHashData' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(111,31): warning C4013: 'CryptGetHashParam' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(169,24): warning C4013: 'CryptDestroyKey' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(188,26): warning C4013: 'CryptEncrypt' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(203,26): warning C4013: 'CryptDecrypt' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(255,36): warning C4013: 'CryptImportKey' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(265,34): warning C4013: 'CryptSetKeyParam' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(402,39): warning C4013: 'CryptSetHashParam' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(472,32): warning C4013: 'CryptDuplicateHash' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(599,38): warning C4013: 'CryptSignMessage' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(640,41): warning C4013: 'CryptVerifyMessageSignature' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
       Link:
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29333\bin\HostX64\x64\link.exe /ERRORREPORT:QUEUE /OUT:"D:\buildtrees\minizip-ng\x64-uwp-dbg\Debug\libminizip-ng.dll" /INCREMENTAL /NOLOGO "D:\installed\x64-uwp\debug\lib\zlibd.lib" "D:\installed\x64-uwp\debug\lib\bz2d.lib" "D:\installed\x64-uwp\debug\lib\lzmad.lib" "D:\installed\x64-uwp\debug\lib\zstdd.lib" crypt32.lib WindowsApp.lib /MANIFEST /ManifestFile:"minizip-ng.dir\Debug\libminizip-ng.dll.intermediate.manifest" /MANIFESTUAC:NO /DEBUG /PDB:"D:/buildtrees/minizip-ng/x64-uwp-dbg/Debug/libminizip-ng.pdb" /SUBSYSTEM:CONSOLE /OPT:REF /OPT:ICF /TLBID:1 /APPCONTAINER /WINMD:NO /WINMDFILE:"D:\buildtrees\minizip-ng\x64-uwp-dbg\Debug\libminizip-ng.winmd" /DYNAMICBASE /NXCOMPAT /IMPLIB:"D:/buildtrees/minizip-ng/x64-uwp-dbg/Debug/libminizip-ng.lib" /MACHINE:X64  /machine:x64 /nologo /DLL "minizip-ng.dir\Debug\mz_crypt.obj"
         "minizip-ng.dir\Debug\mz_os.obj"
         "minizip-ng.dir\Debug\mz_strm.obj"
         "minizip-ng.dir\Debug\mz_strm_buf.obj"
         "minizip-ng.dir\Debug\mz_strm_mem.obj"
         "minizip-ng.dir\Debug\mz_strm_split.obj"
         "minizip-ng.dir\Debug\mz_zip.obj"
         "minizip-ng.dir\Debug\mz_zip_rw.obj"
         "minizip-ng.dir\Debug\mz_strm_zlib.obj"
         "minizip-ng.dir\Debug\mz_strm_bzip.obj"
         "minizip-ng.dir\Debug\mz_strm_lzma.obj"
         "minizip-ng.dir\Debug\mz_strm_zstd.obj"
         "minizip-ng.dir\Debug\mz_os_win32.obj"
         "minizip-ng.dir\Debug\mz_strm_os_win32.obj"
         "minizip-ng.dir\Debug\mz_crypt_win32.obj"
         "minizip-ng.dir\Debug\mz_strm_pkcrypt.obj"
         "minizip-ng.dir\Debug\mz_strm_wzaes.obj"
         "minizip-ng.dir\Debug\mz_compat.obj"
     4>LINK : warning LNK4075: ignoring '/INCREMENTAL' due to '/OPT:ICF' specification [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
            Creating library D:/buildtrees/minizip-ng/x64-uwp-dbg/Debug/libminizip-ng.lib and object D:/buildtrees/minizip-ng/x64-uwp-dbg/Debug/libminizip-ng.exp
     4>mz_crypt_win32.obj : error LNK2019: unresolved external symbol CryptAcquireContext referenced in function mz_crypt_aes_set_key [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>mz_crypt_win32.obj : error LNK2019: unresolved external symbol CryptDuplicateHash referenced in function mz_crypt_hmac_copy [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>D:\buildtrees\minizip-ng\x64-uwp-dbg\Debug\libminizip-ng.dll : fatal error LNK1120: 2 unresolved externals [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
     4>Done Building Project "D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj" (default targets) -- FAILED.
     3>Done Building Project "D:\buildtrees\minizip-ng\x64-uwp-dbg\ALL_BUILD.vcxproj" (default targets) -- FAILED.
     1>Done Building Project "D:\buildtrees\minizip-ng\x64-uwp-dbg\install.vcxproj" (default targets) -- FAILED.

Build FAILED.

       "D:\buildtrees\minizip-ng\x64-uwp-dbg\install.vcxproj" (default target) (1) ->
       "D:\buildtrees\minizip-ng\x64-uwp-dbg\ALL_BUILD.vcxproj" (default target) (3) ->
       "D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj" (default target) (4) ->
       (ClCompile target) -> 
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(25,33): warning C4013: 'CryptAcquireContext' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(27,32): warning C4013: 'CryptGenRandom' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(50,25): warning C4013: 'CryptDestroyHash' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(80,33): warning C4013: 'CryptCreateHash' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(96,27): warning C4013: 'CryptHashData' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(111,31): warning C4013: 'CryptGetHashParam' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(169,24): warning C4013: 'CryptDestroyKey' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(188,26): warning C4013: 'CryptEncrypt' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(203,26): warning C4013: 'CryptDecrypt' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(255,36): warning C4013: 'CryptImportKey' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(265,34): warning C4013: 'CryptSetKeyParam' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(402,39): warning C4013: 'CryptSetHashParam' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(472,32): warning C4013: 'CryptDuplicateHash' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(599,38): warning C4013: 'CryptSignMessage' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\src\3.0.1-4535e59d51.clean\mz_crypt_win32.c(640,41): warning C4013: 'CryptVerifyMessageSignature' undefined; assuming extern returning int [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]


       "D:\buildtrees\minizip-ng\x64-uwp-dbg\install.vcxproj" (default target) (1) ->
       "D:\buildtrees\minizip-ng\x64-uwp-dbg\ALL_BUILD.vcxproj" (default target) (3) ->
       "D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj" (default target) (4) ->
       (Link target) -> 
         LINK : warning LNK4075: ignoring '/INCREMENTAL' due to '/OPT:ICF' specification [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]


       "D:\buildtrees\minizip-ng\x64-uwp-dbg\install.vcxproj" (default target) (1) ->
       "D:\buildtrees\minizip-ng\x64-uwp-dbg\ALL_BUILD.vcxproj" (default target) (3) ->
       "D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj" (default target) (4) ->
       (Link target) -> 
         mz_crypt_win32.obj : error LNK2019: unresolved external symbol CryptAcquireContext referenced in function mz_crypt_aes_set_key [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         mz_crypt_win32.obj : error LNK2019: unresolved external symbol CryptDuplicateHash referenced in function mz_crypt_hmac_copy [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]
         D:\buildtrees\minizip-ng\x64-uwp-dbg\Debug\libminizip-ng.dll : fatal error LNK1120: 2 unresolved externals [D:\buildtrees\minizip-ng\x64-uwp-dbg\minizip-ng.vcxproj]

    16 Warning(s)
    3 Error(s)

Time Elapsed 00:00:01.79

```

nmoinvaz replied:
Seems to be the difference between Crypt and BCrypt, never even heard of BCrypt. That would be something good to add into minizip-ng one day.

prateek9623 replied:
crypto missing?

nmoinvaz replied:
All the crypto functions would need to be prefixed with `B`.

prateek9623 commented:
@nmoinvaz this https://github.com/oswjk/vcpkg-minizip-fork/blob/master/CONTROL  was a previous attempt to add vcpkg port, as you can see here there are many features in this port. so, what do think, who many features should we support?

nmoinvaz commented:
I am somewhat of the opinion that it should support most of the features for completeness at least the ones for turning on/off compression libraries. But how far you want to take it I would leave at your discretion.

prateek9623 commented:
> I am somewhat of the opinion that it should support most of the features for completeness at least the ones for turning on/off compression libraries. But how far you want to take it I would leave at your discretion.

am fine with this for now, may be add support for zlib-ng

PhoebeHui commented on ports/minizip-ng/vcpkg.json:
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "minizip-ng",
+  "version-string": "3.0.1",
```
```suggestion
  "version": "3.0.1",
```

PhoebeHui commented on ports/minizip-ng/portfile.cmake:
```diff
@@ -0,0 +1,24 @@
+vcpkg_copy_pdbs()
+
+vcpkg_fixup_pkgconfig()
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/minizip-ng TARGET_PATH share/minizip-ng)
```
This function has been deprecated, could you please use [vcpkg_cmake_config_fixup](https://github.com/Microsoft/vcpkg/blob/master/docs/maintainers/ports/vcpkg-cmake-config/vcpkg_cmake_config_fixup.md) instead?

PhoebeHui commented on ports/minizip-ng/portfile.cmake:
```diff
@@ -0,0 +1,24 @@
+    OPTIONS -DMZ_PROJECT_SUFFIX:STRING=-ng
+)
+
+vcpkg_install_cmake()
```
This function has been deprecated, could you please use [vcpkg_cmake_install](https://github.com/Microsoft/vcpkg/blob/master/docs/maintainers/ports/vcpkg-cmake/vcpkg_cmake_install.md) instead?

prateek9623 replied:
for this i also needed to change vcpkg_configure_cmake to vcpkg_cmake_configure, but after this it was failing, giving an error while configuration that it was unable to create  "zip.h" and "unzip.h". So, I assumed it was unable to create it because of some conflict so, I tried adding the DISABLE_PARALLEL_CONFIGURE flag and it started working. Is there any other way to resolve this without using the DISABLE_PARALLEL_CONFIGURE flag?

PhoebeHui replied:
Confirmed, I also reproed this issue locally, @prateek9623, thanks for pointing out this!

@strega-nil, what's the difference between vcpkg_configure_cmake and vcpkg_cmake_configure for DISABLE_PARALLEL_CONFIGURE behavior?
```
CMake Error at CMakeLists.txt:632 (file):
  file attempted to write a file:
  E:/vcpkg/17101/vcpkg/buildtrees/minizip-ng/src/3.0.1-4535e59d51/zip.h into
  a source directory.
```

PhoebeHui replied:
@prateek9623, passing DISABLE_PARALLEL_CONFIGURE is correct solution here, a cmake buildsystem that writes to the source directory must _always_ pass DISABLE_PARALLEL_CONFIGURE, enable configure in parallel is not a huge speedup, so we do not need to worry about it. It only valid when one doesn't write to the source directory.

PhoebeHui reviewed (changes requested):
@prateek9623, thanks for the PR!

PhoebeHui commented on ports/minizip-ng/portfile.cmake:
```diff
@@ -0,0 +1,25 @@
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO zlib-ng/minizip-ng
+    REF 3.0.1
+    SHA512 98c9bdcea79a88a2dd69cec6c49f8565edf78ab9cddbf0e85e08b049b300b187f176bf57d5a894bf777bec0a097e46ecc05f78dab9cd5726fd473ffd8718dce0
```
```suggestion
    SHA512 98c9bdcea79a88a2dd69cec6c49f8565edf78ab9cddbf0e85e08b049b300b187f176bf57d5a894bf777bec0a097e46ecc05f78dab9cd5726fd473ffd8718dce0
    HEAD_REF master
```

PhoebeHui replied:
LGTM if update this one.

prateek9623 replied:
done

prateek9623 committed:
[celero] Updated to Celero v2.8.0 (#16802)

* [celero] Updated to Celero v2.8.0

* [celero] Improving static linking support in VCPKG.

Celero issue #154 is tracking a fix.
https://github.com/DigitalInBlue/Celero/issues/154

* [celero] Updated to v2.8.1

* [celero] Updated version for VCPKG.

* [celero] Updates to better support VCPKG.

* [celero] Updated versions.

* Update celero.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>
